packages/tooltip/src/vaadin-tooltip-mixin.js

```diff
@@ -668,7 +685,22 @@ export const TooltipMixin = (superClass) =>
 
     /** @private */
     __updateContent() {
-      this.__contentNode.textContent = typeof this.generator === 'function' ? this.generator(this.context) : this.text;
+      const content = typeof this.generator === 'function' ? this.generator(this.context) : this.text;
+
+      if (this.contentType === 'markdown' && content) {
+        this.__importMarkdownHelpers().then((helpers) => {
```

Comment:
nit: could also use `async` / `await` here, then we don't need `__updateContentFinished()` method.